File: mvp-widget-sample/src/main/java/ru/surfstudio/android/mvpwidget/sample/interactor/ui/base/configurator/ActivityScreenConfigurator.java
package ru.surfstudio.android.mvpwidget.sample.interactor.ui.base.configurator;

import android.content.Intent;

import ru.surfstudio.android.core.mvp.configurator.BaseActivityViewConfigurator;
import ru.surfstudio.android.mvpwidget.sample.interactor.app.App;
import ru.surfstudio.android.mvpwidget.sample.interactor.app.dagger.AppComponent;
import ru.surfstudio.android.mvpwidget.sample.interactor.ui.base.dagger.activity.ActivityComponent;
import ru.surfstudio.android.mvpwidget.sample.interactor.ui.base.dagger.activity.ActivityModule;
import ru.surfstudio.android.mvpwidget.sample.interactor.ui.base.dagger.screen.ActivityScreenModule;
import ru.surfstudio.android.mvpwidget.sample.interactor.ui.base.dagger.activity.DaggerActivityComponent;

/**
 * Базовый конфигуратор для экрана, основанного на активити
 */

public abstract class ActivityScreenConfigurator
        extends BaseActivityViewConfigurator<AppComponent, ActivityComponent, ActivityScreenModule> {

    public ActivityScreenConfigurator(Intent intent) {
        super(intent);
    }

    @Override
    protected ActivityComponent createActivityComponent(AppComponent parentComponent) {
        return DaggerActivityComponent.builder()
                .appComponent(parentComponent)
                .activityModule(new ActivityModule(getPersistentScope()))
                .build();
    }

    @Override
    protected AppComponent getParentComponent() {
        return ((App) (getTargetActivity()).getApplicationContext()).getAppComponent();
    }

    @Override
    protected ActivityScreenModule getActivityScreenModule() {
        return new ActivityScreenModule();
    }
}

File: template-multimodule/app-injector/src/main/java/ru/surfstudio/standard/app_injector/ui/configurator/ActivityScreenConfigurator.kt
package ru.surfstudio.standard.app_injector.ui.configurator

import android.app.Activity
import android.content.Intent
import ru.surfstudio.android.core.mvp.activity.CoreActivityView
import ru.surfstudio.android.core.mvp.configurator.BaseActivityViewConfigurator
import ru.surfstudio.standard.app_injector.ui.ActivityComponent
import ru.surfstudio.standard.app_injector.ui.ActivityModule
import ru.surfstudio.standard.app_injector.ui.DaggerActivityComponent
import ru.surfstudio.standard.app_injector.ui.screen.ActivityScreenModule

/**
 * Базовый конфигуратор для экрана, основанного на [Activity]
 */
abstract class ActivityScreenConfigurator(intent: Intent) :
        BaseActivityViewConfigurator<ru.surfstudio.standard.app_injector.AppComponent, ActivityComponent, ActivityScreenModule>(intent) {

    override fun createActivityComponent(parentComponent: ru.surfstudio.standard.app_injector.AppComponent): ActivityComponent {
        return DaggerActivityComponent.builder()
                .appComponent(parentComponent)
                .activityModule(ActivityModule(persistentScope))
                .build()
    }

    override fun getParentComponent(): ru.surfstudio.standard.app_injector.AppComponent {
        return (getTargetActivity<CoreActivityView>().applicationContext as ru.surfstudio.standard.app_injector.App).appComponent
    }

    override fun getActivityScreenModule(): ActivityScreenModule {
        return ActivityScreenModule(persistentScope)
    }
}